pkg/auth/providers/common/ldap/ldap_util.go

```diff
@@ -170,7 +170,7 @@ func GetGroupSearchAttributesForLDAP(ObjectClass string, config *v3.LdapConfig)
 	return groupSeachAttributes
 }
 
-func AuthenticateServiceAccountUser(serviceAccountPassword string, serviceAccountUsername string, defaultLoginDomain string, lConn *ldapv3.Conn) error {
+func AuthenticateServiceAccountUser(serviceAccountPassword string, serviceAccountUsername string, defaultLoginDomain string, lConn ldapv3.Client) error {
```

Comment:
This may be my own ignorance of the language at work here, but passing in the `ldapv3.Client` by **value** looks strange. What happens if we need to authenticate two service accounts in a row, and we pass the same client (from the call site) in twice? Does the state get synchronized in some way that I'm missing?

Reply:
d'oh, that's an _interface_. Ignore me, this is fine.